### PR TITLE
Admin interface: buttons to show/hide action buttons [OJS 3.1.2+]

### PR DESCRIPTION
**Pull request by retostauffer**
Hy @NateWr

Took me a while to find some time to come back to this issue (#4018). I opened a new pull request as I could not figure out how to work on the previous one.

Started from scratch (on current master branch), implemented the "hide button" draft again and did some fine tuning. I'll put some comments in the old pull request, however, feel free to close the previous one (#4018).

I'll also include some comments below.

### CODE REVIEW

File: js/controllers/linkAction/LinkActionHideButtons.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	 * @extends $.pkp.controllers
+	 */
+	$.pkp.controllers.LinkAction_LinkActionHideButtons = function() {
+
+		alert('include script code here, somehow?')
+
+	};
+
+/** @param {jQuery} $ jQuery closure. */
+}(jQuery));
```
Comment (retostauffer) [inline js/controllers/linkAction/LinkActionHideButtons.js]: **Note** this is just how I think the java script part might be included? You told me that you'll help me with that if needed, seems it is needed *g*. I think I figured out how how it basically works, but could not figure out how to get it running. I think I have to compile the js library (npm?). Where should this script be located? I just put it into "controllers/linkAction/LinkActionHideButtons" (t.b.d.).

File: locale/en_US/editor.xml [hunk 2]
```diff
@@ -12,6 +12,18 @@
...
+	<message key="editor.submission.workflowDecision.Submission.accepted">Submission accepted.</message>
+	<message key="editor.submission.workflowDecision.Submission.published">Submission has been published.</message>
+	<message key="editor.submission.workflowDecision.Submission.review">Sent into review, not yet sent to reviewers.</message>
+	<message key="editor.submission.workflowDecision.Submission.external">Sent out to reviewers (underway).</message>
+	<message key="editor.submission.workflowDecision.Submission.revisionrequired">Decision recorded: revision required.</message>
+	<message key="editor.submission.workflowDecision.Submission.copyediting">Sent to copyediting.</message>
+	<message key="editor.submission.workflowDecision.Submission.production">Sent to production.</message>
+	<message key="editor.submission.workflowDecision.Submission.resubmit">Decision recorded: resubmit for review.</message>
+	<message key="editor.submission.workflowDecision.Submission.underreview">Submission set to review.</message>
+	<message key="editor.submission.workflowDecision.changeDecision">Change decision</message>
```
Comment (retostauffer) [inline locale/en_US/editor.xml]: Additional messages, en_US only.

File: pages/workflow/PKPWorkflowHandler.inc.php [hunk 3]
```diff
@@ -328,12 +329,59 @@ function editorDecisionActions($args, $request) {
...
+			// If submission published (status 3)
+			case STATUS_PUBLISHED:
+				$lastDecision = 'editor.submission.workflowDecision.Submission.published';
+				break;
+			// If submission declined (status 4)
+			case STATUS_DECLINED:
+				$lastDecision = 'editor.submission.workflowDecision.Submission.declined';
+				break;
+		}
+
```
Comment (retostauffer) [inline pages/workflow/PKPWorkflowHandler.inc.php]: Revamped the logic. I went trough a range of possible states in one of my test installations. So far it looked quite good.

File: styles/pages/workflow.less [hunk 4]
```diff
@@ -134,14 +134,55 @@
...
+	.show_extras:before {
+		&:extend(.pkp_caret);
+		&:extend(.pkp_caret_right);
+		margin-top: -4px;
+	}
+
+	.hide_extras:before {
+		&:extend(.pkp_caret);
+		&:extend(.pkp_caret_down);
+	}
```
Comment (retostauffer) [inline styles/pages/workflow.less]: Tried to mimic the `.show_extras` and `.hide_extras` defined in `.pkp_controllers_grid`, however, was not successful. Thus this is, in large parts, a copy of `.pkp_controllers_grid .show_extras` and `.pkp_controllers_grid .hide_extras`. There might be a nicer way, but I was not able to figure out what the best way would be.

File: templates/workflow/editorialLinkActions.tpl [hunk 5]
```diff
@@ -7,6 +7,17 @@
...
+{if (!empty($lastDecision))}
+	<script type="text/javascript">
+		// Expected to include the script somehow like this?
+		//$(function() {ldelim}
+		//	$('.pkp_workflow_decided').pkpHandler(
+		//		'$.pkp.controllers.LinkAction_LinkActionHideButtons'
+		//	);
+		//{rdelim});
+	</script>
+{/if}
```
Comment (retostauffer) [inline templates/workflow/editorialLinkActions.tpl]: As in https://github.com/pkp/pkp-lib/pull/4583/commits/9eab04a1b14de44333d409fde7b5ad5329db236a#diff-bc97fc94e30294769a544cd4a2468a95 this is just a draft where I think the js code should/could be placed Help appreciated and needed! :)

File: templates/workflow/editorialLinkActions.tpl [hunk 6]
```diff
@@ -24,12 +35,41 @@
...
+					if ( $(theul).css("display") == "none" ) {ldelim}
+						$(this).switchClass("show_extras", "hide_extras");
+						$(theul).fadeIn();
+					{rdelim} else {ldelim}
+						$(this).switchClass("hide_extras", "show_extras");
+						$(theul).fadeOut();
+					{rdelim}
+				{rdelim});
+			{rdelim});
+			</script>
```
Comment (retostauffer) [inline templates/workflow/editorialLinkActions.tpl]: Currently, the script part is still included in the template file.

**Comment by NateWr** [comment]
Thanks @retostauffer. I apologize (again) for my slow follow-up on this. The 3.1.2 release turned into a real stretch and I've been flat out with work on a tight deadline all year. That means stuff like this gets pushed back.

I've made a reminder to myself to look at this in two weeks, when I hope to be done with my current short-deadline task. If you don't hear from me after that, please do remind me. I do want to get this in, but it's something I need to take a closer look at and that means it's harder to squeeze it in here and there.

**Comment by NateWr** [comment]
Closing this PR in favor of the new one here: https://github.com/pkp/pkp-lib/pull/4641

**Pull request closed without merging**